website/src/components/Stats/Stats.components.tsx

```diff
@@ -249,10 +267,30 @@ export const MessageTreeStateStatsStacked = ({ stat }: MessageTreeStateStatsProp
   );
 };
 
+// Returns the color for the bar chart. If the stat is a message tree state, it returns
+// an array of colors depending on the state. Otherwise, it returns an empty string, so
+// that the default color is used.
+function getBackgroundColor(stat, data) {
+  if (stat.name === "message_trees_by_state") {
+    return data.map((item) => stateColors[item]);
+  }
+  return "";
+}
+
 export const Chart = ({ stat, titleFn, type: Component }: ChartProps) => {
-  const data = Object.keys(stat.stats);
+  let data = Object.keys(stat.stats);
   const { t } = useTranslation("stats");
   const color = useColorModeValue(colors.light.text, colors.dark.text);
+
+  if (stat.name === "message_trees_by_state") {
+    // Order data by interesting state
+    data = data.sort((a, b) => {
```

Comment:
You dont need to reassign the `data` variable. `sort` mutate the original array

Reply:
I didn't know that, thank you. Fixed!